WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderCommentListActivity.java

```diff
@@ -189,43 +183,38 @@ public void onClick(View v) {
         }
     };
 
+    // to do a complete refresh we need to get updated post and new comments
     private void updatePostAndComments() {
-        //to do a complete refresh we need to get updated post and new comments
         ReaderPostActions.updatePost(mPost, new ReaderActions.UpdateResultListener() {
             @Override
             public void onUpdateResult(ReaderActions.UpdateResult result) {
-                if (isFinishing()) {
-                    return;
-                }
-
-                if (result.isNewOrChanged()) {
-                    getCommentAdapter().setPost(mPost); //pass updated post to the adapter
-                    ReaderCommentTable.purgeCommentsForPost(mBlogId, mPostId); //clear all the previous comments
-                    updateComments(false, false); //load first page of comments
-                } else {
-                    setRefreshing(false);
+                if (!isFinishing() && result.isNewOrChanged()) {
+                    // get the updated post and pass it to the adapter
+                    ReaderPost post = ReaderPostTable.getBlogPost(mBlogId, mPostId, false);
+                    if (post != null) {
+                        getCommentAdapter().setPost(post);
+                        mPost = post;
+                    }
                 }
             }
         });
+
+        // clear all the previous comments, then load the first page of comments
+        ReaderCommentTable.purgeCommentsForPost(mBlogId, mPostId);
```

Comment:
@daniloercoli While looking into this I discovered another issue with the incorrect comment count being shown. I'm going to put this PR on hold until I have time to track down that problem.